src/main/java/frc/team2412/robot/util/AbsoluteSwerveModule.java

```diff
@@ -0,0 +1,42 @@
+package frc.team2412.robot.util;
+
+import com.ctre.phoenix.ErrorCode;
+import com.ctre.phoenix.sensors.CANCoder;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+
+public class AbsoluteSwerveModule extends SwerveModule {
+
+    CANCoder absoluteEncoder;
+
+    public AbsoluteSwerveModule(int driveMotorPort, int turnMotorPort, int turnEncoderPort, double turnOffset,
+            String canbus) {
+        super(driveMotorPort, turnMotorPort, canbus);
+        configAbsoluteEncoder(turnEncoderPort, turnOffset, canbus);
+
+    }
+
+    private void configAbsoluteEncoder(int turnEncoderPort, double turnOffset, String canbus) {
+
+        absoluteEncoder = new CANCoder(turnEncoderPort, canbus);
+        ErrorCode encoderCreatedCheck = absoluteEncoder.configMagnetOffset(turnOffset, 500);
+
+        // I feel like this will break but 🤷‍♂️
```

Comment:
If you want to be more certain it doesn't break, you could a counter and check the number of recursive calls... although at that point you might as well convert it into a loop.

It's basically a choice between "only a module broke" and "the entire robot crashes" - although practically speaking, we may not be able to do much with a broken swerve module anyway...